proofgeneral/narya.el

```diff
@@ -267,6 +268,37 @@ it won't try to duplicate our work."
 	"Show the goal and context for all open holes."
 	(interactive)
 	(proof-shell-invisible-command "show holes"))
+	
+(defun narya-solve-hole ()
+  "Solve the current hole with a user-provided term."
+  (interactive)
+  (let ((hole-overlay (car (seq-filter (lambda (ovl)
+                                         (eq (get-char-property (point) 'narya-hole)
+                                             (overlay-get ovl 'narya-hole)))
+                                       narya-hole-overlays))))
```

Comment:
Another option here would be to iterate only through the overlays at point with `overlays-in`, looking for one that has a `narya-hole` property.  I don't know which would be better, just thought I would mention it.

Reply:
If you mean the following
```emacs
(let ((hole-overlay (car (seq-filter (lambda (ovl)
                                       (overlay-get ovl 'narya-hole))
                                     (overlays-in (point) (point))))))
```
It didn't work because it does not capture a hole overlay. But I might have mistyped it. 

However, the following worked
```emacs
(let ((hole-overlay (car (seq-filter (lambda (ovl)
                                         (overlay-get ovl 'narya-hole))
                                       (overlays-at (point))))))
```

Reply:
Ah, yes, that makes sense -- `overlays-in` requires at least one character to be contained in both the overlay and the region supplied, whereas the region from `point` to `point` is always empty.  So `overlays-at` is correct.